extensions/mssql/src/util/dataCache.ts

```diff
@@ -0,0 +1,40 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+export class DataCache<T> {
+
+	millisecondsToLive: number;
+	getValueFunction: (...args: any[]) => T;
+	cache: T;
+	fetchDate: Date;
+
+	constructor(getValueFunction: (...args: any[]) => T, secondsToLive: number) {
+		this.millisecondsToLive = secondsToLive * 1000;
+		this.getValueFunction = getValueFunction;
+		this.cache = undefined;
+		this.fetchDate = new Date(0);
+	}
+
+	public isCacheExpired(): boolean {
+		return (this.fetchDate.getTime() + this.millisecondsToLive) < new Date().getTime();
+	}
+
+	public getData(...args: any[]): T {
+		if (!this.cache || this.isCacheExpired()) {
+			console.log('expired - fetching new data');
```

Comment:
Please be more descriptive here about what's expired - since this is going in the console it's hard to get context from messages like this

Reply:
Oh yea, these console logs were for manual testing and debugging and are not necessary. I have removed them altogether.